### ISSUE TITLE
Value.Check() fails if optional property missing from object.

### ISSUE THREAD

**Issue by waikikamoukow** [sinclairzx81/sidewinder#19]
Currently Value.Check() will fail if the schema is of type Object and the supplied value is missing any properties from the Object schema, even if those properties are Optional.

Example:
```
const schema = Type.Object({testProperty: Type.Optional(Type.String())})
const check = Value.Check(schema, {}) // Will return false
const nextCheck = Value.Check(schema, {testProperty: ''}) // Will return true
```
This does not align with the behaviour of Value.Upcast(), which will only create non-optional properties. So this does not work:
```
const schema = Type.Object({testProperty: Type.Optional(Type.String())})
const upcast = Value.Upcast(schema, null)
const check = Value.Check(schema, upcast) // Will return false, despite upcast
```
But this does:
```
const schema = Type.Object({testProperty: Type.String()})
const upcast = Value.Upcast(schema, null)
const check = Value.Check(schema, upcast) // Will return true
```

**Comment by sinclairzx81** [comment]
@waikikamoukow Hi Jacques! I've pushed a fix for this on `0.9.2`. Just remember to run a full npm update on all dependencies as Sidewinder increments the revision across all packages whenever one package changes (to keep things in sync). The update also merges some of the latest from TypeBox (which includes more exhaustive type checks), and also includes the new `Value.Errors(T, value)` that returns a list of errors for any object that fails `Value.Check(T, value)`.  https://github.com/sinclairzx81/typebox#compiler

Also, just a heads up. There is actually a few changes planned for the `Value.*` module in future revisions. The changes mostly have to do with refactoring imports to be inline with TypeBox.

### Current

The current package is imported on `@sidewinder/value`. 

```typescript
import { Value } from '@sidewinder/value'
import { Type } from '@sidewinder/type'
```

### Future

Future revisions will move the value package underneath `@sidewinder/type`. This reduces the number of installable dependencies while still allowing bundlers to tree shake.

```typescript
import { Value } from '@sidewinder/type/value'
import { Type } from '@sidewinder/type'
```

Also, `Value.Upcast()` will be renamed to `Value.Cast()` in future releases (inline with TypeBox). These updates are not planned for a while, but will likely go through somewhere close to the end of the year. 

Let me know if these updates fix the issue.
Cheers
S

**Comment by sinclairzx81** [comment]
@waikikamoukow Hey, might close off this issue. Have added a couple of tests around this which you can preview here https://github.com/sinclairzx81/sidewinder/blob/0dcda857c177d496c5963c593a1b3d43f877a9c0/tests/value/upcast/object.ts#L113-L153

Hopefully this solves the issue. But feel free to ping on this thread if you still see a bug. 
Cheers
S
